.travis.yml

```diff
@@ -1,16 +1,21 @@
 language: scala
 scala: 2.12.4
 jdk: oraclejdk8
+sudo: false
+dist: trusty
 
 env:
   global:
     - TRAVIS_NODE_VERSION="8"
-    - TRAVIS_YARN_VERSION="1.1.0"
+    - TRAVIS_YARN_VERSION="1.3.2"
 
 matrix:
   include:
     - script: sbt ci-test
       env: CI_PUBLISH=true
+      deploy:
+        provider: script
+        script: bash bin/ci-publish.sh
```

Comment:
I have never used deploy before, what is the benefit of deploy compared to after_success or build stages? One thing that has bugged me about after_success is that failures are silent and don't trigger a build error (no email notification).

Reply:
`deploy` supports conditional triggers based on wether the commit has a tag or is a specific branch, we handle that in the publish script, however, I just checked and if the script fails it will fail the build, so I think it still makes sense.

Reply:
If this fails the build on error then it's a big win, very annoying when after_success fails silently on publish error